src/Tools/Test Framework/Test Runner/src/TestSuiteMgt.Codeunit.al

```diff
@@ -585,6 +613,7 @@ codeunit 130456 "Test Suite Mgt."
         TestMethodLine."Test Codeunit" := CodeunitMetadata.ID;
         TestMethodLine.Validate("Line Type", TestMethodLine."Line Type"::Codeunit);
         TestMethodLine.Name := CodeunitMetadata.Name;
+        TestMethodLine.Validate("Test Type", CodeunitMetadata.TestType);
```

Comment:
This does not seem correct. The type should be on the suite. 
We should not call AddTestMethod if the type is not matching the type on the suite.